hibernate-core/src/main/java/org/hibernate/engine/jdbc/env/internal/JdbcEnvironmentInitiator.java

```diff
@@ -148,12 +149,15 @@ else if ( explicitDialectConfiguration( explicitDatabaseName, configurationValue
 			jdbcEnvironment = getJdbcEnvironmentWithDefaults( configurationValues, registry, dialectFactory );
 		}
 
-		logDbInfo( registry, jdbcEnvironment );
+		logDbInfo( isJdbcMetadataAccessAllowed, registry, jdbcEnvironment );
```

Comment:
@jrenaat why not just move the call to `logDbInfo()` inside the first block of the `if`? That's simpler and clearer, no?